conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,643 @@
+    error err2 = err1;
+    io:println(err2 is E1); // @output true
+
+    E err3 = error("message", code = 100, no = 200);
+    E3 err4 = err3;
+    io:println(err4 is E); // @output true
+}
+
+Test-Case: output
+Description: Test nested error constructor.
+Labels: error-constructor-expr, error-type, error-type-parameter, error:cause, error:detail, error:message,
+        if-else-stmt, is-expr, method-call-expr, module-type-defn, optional-type, record-type, string
+
+type E error<record {|string message?; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("message 1", error E("message 2", message = "ERROR!", fatal = true));
+    io:println(err1.message()); // @output message 1
+    io:println(err1.detail()); // @output {}
+
+    error? cause1 = err1.cause();
+    io:println(cause1 is error); // @output true
+    if cause1 is error {
+        io:println(cause1.message()); // @output message 2
+        io:println(cause1.detail()); // @output {"message":"ERROR!","fatal":true}
+    }
+
+    error err2 = error("message 1", error E("message 2", error("message 3", error("message 4", fatal = false)), message = "ERROR!", fatal = true), errorCode = "BCE001");
+    io:println(err2.message()); // @output message 1
+    io:println(err2.detail()); // @output {"errorCode":"BCE001"}
+
+    error? cause2 = err2.cause();
+    io:println(cause2 is error); // @output true
+    if cause2 is error {
+        io:println(cause2.message()); // @output message 2
+        io:println(cause2.detail()); // @output {"message":"ERROR!","fatal":true}
+
+        error? cause3 = cause2.cause();
+        io:println(cause3 is error); // @output true
+        if cause3 is error {
+            io:println(cause3.message()); // @output message 3
+            io:println(cause3.detail()); // @output {}
+
+            error? cause4 = cause3.cause();
+            io:println(cause4 is error); // @output true
+            if cause4 is error {
+                io:println(cause4.message()); // @output message 4
+                io:println(cause4.detail()); // @output {"fatal":false}
+            }
+        }
+    }
+}
+
+Test-Case: output
+Description: Test error constructor when the contextually expected type is a union.
+Labels: error-constructor-expr, error-type, error:stackTrace, function-call-expr, method-call-expr, optional-type
+
+function init() {
+    error? fooResult = foo();
+
+    io:println(fooResult is error); // @output true
+    if fooResult is error {
+        io:println(fooResult.stackTrace()); // @output [callableName: baz  fileName: int_620.bal lineNumber: 20,callableName: bar  fileName: int_620.bal lineNumber: 16,callableName: foo  fileName: int_620.bal lineNumber: 12,callableName: init  fileName: int_620.bal lineNumber: 3]
+    }
+}
+
+function foo() returns error? {
+    return bar();
+}
+
+function bar() returns error? {
+    return baz();
+}
+
+function baz() returns error? {
+    return error("error message", fatal = false);
+}
```

Comment:
Shall we also add a few tests to cover `error constructor expr` at module level? Others LGTM.